include/boost/geometry/algorithms/discrete_frechet_distance.hpp

```diff
@@ -0,0 +1,221 @@
+        boost::geometry::detail::throw_on_empty_input(ls2);
+
+        size_type1 const a = boost::size(ls1);
+        size_type2 const b = boost::size(ls2);
+
+
+        //Coupling Matrix CoupMat(a,b,-1);
+        coup_mat<size_type1,size_type2,result_type>  coup_matrix(a,b);
+
+        result_type const not_feasible = -100;
+        //findin the Coupling Measure
+        for(size_type1 i=0;i<a;i++)
+        {
+            for(size_type2 j=0;j<b;j++)
+            {
+                result_type dis = geometry::distance(range::at(ls1,i),range::at(ls2,j),strategy);
+                if(i==0 && j==0)
+                    coup_matrix(i,j)= dis;
+                else if(i==0 && j>0)
+                    coup_matrix(i,j)=
+                    (std::max)(coup_matrix(i,j-1),dis);
+                else if(i>0 && j==0)
+                    coup_matrix(i,j)=
+                    (std::max)(coup_matrix(i-1,j),dis);
+                else if(i>0 && j>0)
+                    coup_matrix(i,j)=
+                    (std::max)((std::min)(coup_matrix(i,j-1),(std::min)(coup_matrix(i-1,j),coup_matrix(i-1,j-1))),dis);
+                else
+                    coup_matrix(i,j)=not_feasible;
+            }
+        }
+
+        #ifdef BOOST_GEOMETRY_DEBUG_FRECHET_DISTANCE
+        //Print CoupLing Matrix
+        for(size_type i = 0; i <a; i++)
+        {
+            for(size_type j = 0; j <b; j++)
+            std::cout << coup_matrix(i,j) << " ";
+            std::cout << std::endl;
+        }
+        #endif
+            
+        return coup_matrix(a-1,b-1);
+    }
+};
+
+}} // namespace detail::frechet_distance
+#endif // DOXYGEN_NO_DETAIL
+
+#ifndef DOXYGEN_NO_DISPATCH
+namespace dispatch
+{
+template
+<
+    typename Geometry1,
+    typename Geometry2,
+    typename Tag1 = typename tag<Geometry1>::type,
+    typename Tag2 = typename tag<Geometry2>::type
+>
+struct discrete_frechet_distance : not_implemented<Tag1, Tag2>
+{};
+template <typename Linestring1, typename Linestring2>
+struct discrete_frechet_distance<Linestring1,Linestring2,linestring_tag,linestring_tag>
+    : detail::discrete_frechet_distance::linestring_linestring
+{};
+
+} // namespace dispatch
+#endif // DOXYGEN_NO_DISPATCH
+
+
+/*!
+\brief calculate discrete frechet distance between two geometries using specified strategy
+\ingroup discrete_frechet_distance
+\tparam Geometry1 \tparam_geometry
+\tparam Geometry2 \tparam_geometry
+\tparam Strategy A type fulfilling a DistanceStrategy concept
+\param geometry1 Input geometry
+\param geometry2 Input geometry
+\param strategy Distance strategy to be used to calculate Pt-Pt distance
+
+\qbk{distinguish,with strategy}
+\qbk{[include reference/algorithms/discrete_frechet_distance.qbk]}
+ 
+\qbk{
+[heading Available Strategies]
+\* [link geometry.reference.strategies.strategy_distance_pythagoras Pythagoras (cartesian)]
+\* [link geometry.reference.strategies.strategy_distance_haversine Haversine (spherical)]
+
```

Comment:
What do you mean by that? The algorithm doesn't compile? The result is incorrect? Or documentation cannot be built?

Right now geographic strategies are not mentioned at all in the documentation I think. So if we wanted to add link here we should first document geographic Pt-Pt distance strategy.

Reply:
@yaghya It would be good to add some tests using the other strategies too. `andoyer` is the default so I guess you have to test `vincenty` and `thomas`. 

Regarding documentation what about adding text (without link) to notify the user that geographic strategies can be used even if they are currently undocumented.

Reply:
If you decide to add more tests use:

    strategy::distance::geographic<strategy::andoyer>()
    strategy::distance::geographic<strategy::thomas>()
    strategy::distance::geographic<strategy::vincenty>()

Regarding documentation I think it's a bad idea. If we agree that the current interface for geographic CS will not change the documentation has to be updated consistently. In that case all geographic strategies, srs spheroid and geographic policies (`andoyer`, `thomas`, `vincenty`) has to be documented and links to strategies added in every algorithm, not only here.

Reply:
Added test cases Geographic Strategies and Updated the documentation.

Reply:
This change only makes the docs more confusing, not less.